execution/gethexec/blockchain.go

```diff
@@ -202,13 +204,25 @@ func WriteOrTestChainConfig(chainDb ethdb.Database, config *params.ChainConfig)
 	return nil
 }
 
-func GetBlockChain(chainDb ethdb.Database, cacheConfig *core.CacheConfig, chainConfig *params.ChainConfig, txLookupLimit uint64) (*core.BlockChain, error) {
+func GetBlockChain(
+	chainDb ethdb.Database,
+	cacheConfig *core.CacheConfig,
+	chainConfig *params.ChainConfig,
+	txLookupLimit uint64,
+	tracingConfig json.RawMessage,
```

Comment:
I think you should not need to chainge blockchain at all. However, if you do - seems better to get here *tracing.Hooks that a json.RawMessage.